lib/bolt/config.rb

```diff
@@ -397,35 +416,11 @@ def validate
               "is automatically appended to the modulepath and cannot be configured."
       end
 
-      keys = OPTIONS.keys - %w[plugins plugin_hooks puppetdb]
-      keys.each do |key|
-        next unless Bolt::Util.references?(@data[key])
-        valid_keys = TRANSPORT_CONFIG.keys + %w[plugins plugin_hooks puppetdb]
-        raise Bolt::ValidationError,
-              "Found unsupported key _plugin in config setting #{key}. Plugins are only available in "\
-              "#{valid_keys.join(', ')}."
-      end
-
-      unless concurrency.is_a?(Integer) && concurrency > 0
-        raise Bolt::ValidationError,
-              "Concurrency must be a positive Integer, received #{concurrency.class} #{concurrency}"
-      end
-
-      unless compile_concurrency.is_a?(Integer) && compile_concurrency > 0
-        raise Bolt::ValidationError,
-              "Compile concurrency must be a positive Integer, received #{compile_concurrency.class} "\
-              "#{compile_concurrency}"
-      end
-
       compile_limit = 2 * Etc.nprocessors
       unless compile_concurrency < compile_limit
         raise Bolt::ValidationError, "Compilation is CPU-intensive, set concurrency less than #{compile_limit}"
       end
 
-      unless %w[human json rainbow].include? format
-        raise Bolt::ValidationError, "Unsupported format: '#{format}'"
```

Comment:
Previously this would validate CLI args as well as config options. It seems like most CLI args get caught either by optparse or in later validation, but at least this stacktraces:
```
lucy ~/githubs/bolt ((f8c19af...)) $ be bolt command run whoami -t docker --format 2
bundler: failed to load command: bolt (/home/lucy/githubs/bolt/vendor/bundle/ruby/2.6.0/bin/bolt)
NoMethodError: undefined method `print_head' for nil:NilClass
  /home/lucy/githubs/bolt/lib/bolt/cli.rb:512:in `execute'
  /home/lucy/githubs/bolt/exe/bolt:11:in `<top (required)>'
  /home/lucy/githubs/bolt/vendor/bundle/ruby/2.6.0/bin/bolt:23:in `load'
  /home/lucy/githubs/bolt/vendor/bundle/ruby/2.6.0/bin/bolt:23:in `<top (required)>'

lucy ~/githubs/bolt ((f8c19af...)) $ bolt command run whoami -t docker --format 2
Unsupported format: '2'
```
It seems like the easiest way to address that is to just check CLI options against the schema when we initialize config (I think right after calling  `normalize_overrides`) - that way we still get location for each schema but also validate the CLI.